### PR TITLE
properly check the API during updates if it requires authentication

### LINKED ISSUE #761: Password protected APIs fail to upgrade

### PR DESCRIPTION
**Pull request by rtertiaer**
### What does this change intend to accomplish?
This PR properly checks URLs during the update process in the case where they requrie authentication. This closes #761 .

### Checklist

* [x] Have you tested your changes and ensured they work?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] If applicable, have you updated the documentation/manual?
* [x] If applicable, have you updated the CHANGELOG?
* [x] Does your submission pass linting & tests? You can test on localhost using `./scripts/test`

### CODE REVIEW

File: scripts/configure.py [hunk 1]
```diff
@@ -823,8 +824,29 @@ def _copy_old_config(dest_dir: str) -> None:
       pass
 
 
+def _api_key() -> Union[bool, str]:
```
Comment (linknum23) [inline scripts/configure.py:827]: Its probably a little more pythonic to return Optional[str] here.

**Review by linknum23 — APPROVED** [review]
LGTM, assuming this was tested in an update with and without a key

File: scripts/configure.py (hunk 1 above)
Comment (linknum23) [reply scripts/configure.py:827]: Oddly that change doesn't even effect the downstream logic.

**Comment by rtertiaer** [comment]
Given #812 , I may re-work this idea and instead just modify the auth to not be necesasry if the `Host` is localhost.

**Comment by rtertiaer** [comment]
actually, simply checking the `Host` header invites us to [header smuggling attacks](https://www.intruder.io/research/practical-http-header-smuggling). I'm gonna implement that same methodology for the display server instead.

**Pull request merged**